internal/reader/reader.go

```diff
@@ -64,6 +65,8 @@ func reader(r *kafka.Reader) {
 		msg, err := r.ReadMessage(ctxt)
 		if err != nil {
 			logger.AppLog.Errorf("Error reading off kafka bus err:%v", err)
+			time.Sleep(10 * time.Millisecond)
```

Comment:
What about initial Kafka connect.. I don't see why metric function was crashing here ?

Reply:
TCP connect fails as Kafka pod is still not up, so read operation throws error however, flow continues and it crashes while doing unmarshal of failed read operation.